### PR TITLE
api: order the deposit outpoints by Spend tx input in listspentxs

### PR DESCRIPTION
**Pull request by darosior**
A hack needed for the demo to be removed ASAP.

### CODE REVIEW

File: src/daemon/jsonrpc/api.rs [hunk 1]
```diff
@@ -1123,6 +1123,21 @@ impl RpcApi for RpcImpl {
...
+            // FIXME: a temporary hack for the GUI to be able to use its signer properly. Remove
+            // after the demo release in favour of having the derivation indexes in the PSBT.
+            let mut deposit_outpoints = dep_outpoints.clone();
+            for o in dep_outpoints {
+                let i = db_spend
+                    .psbt
+                    .tx()
+                    .input
+                    .iter()
+                    .position(|i| i.previous_output == o)
```
Comment (edouardparis) [inline src/daemon/jsonrpc/api.rs:1136]: I think it is wrong again, previous output is unvault outpoint not deposit outpoint

Comment (darosior) [reply src/daemon/jsonrpc/api.rs]: Yeah as said on SIgnal, brainfart. Should be good now

**Comment by edouardparis** [comment]
Signature process is tested and setspendtx checked that all the signatures required where here.
ACK a1c60dfadd75c534d1db804e668b7637082e0d32

**Comment by darosior** [comment]
Amended to add a test exercising the new behaviour

**Comment by darosior** [comment]
ACK 2642a7e

**Pull request merged**